src/integrationtests/java/com/aws/greengrass/integrationtests/logmanager/LogManagerTest.java

```diff
@@ -350,4 +350,52 @@ void GIVEN_log_manager_in_errored_state_WHEN_restarted_THEN_logs_upload_is_reatt
         }
         assertEquals(1, allFiles.size());
     }
+
+    //TODO: this test is only for getting some certain level of knowledge of current change uploading active log file
+    // . It will be eventually removed.
+    @Test
+    void GIVEN_user_component_config_with_small_periodic_interval_WHEN_active_logs_included_THEN_logs_are_uploaded_to_cloud()
+            throws Exception {
+        when(cloudWatchLogsClient.putLogEvents(any(PutLogEventsRequest.class)))
+                .thenReturn(PutLogEventsResponse.builder().nextSequenceToken("nextToken").build());
+        logManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.set(true);
+        tempDirectoryPath = Files.createTempDirectory(tempRootDir, "IntegrationTestsTemporaryLogFiles");
+
+        for (int i = 0; i < 5; i++) {
+            createTempFileAndWriteData(tempDirectoryPath, "integTestRandomLogFiles.log_",  "");
+        }
+        createTempFileAndWriteData(tempDirectoryPath, "integTestRandomLogFiles.log",  "");
+
+        setupKernel(tempDirectoryPath, "smallPeriodicIntervalUserComponentConfig.yaml");
+
+        TimeUnit.SECONDS.sleep(30);
```

Comment:
Instead of sleeping it would be nice to have something that checks for a condition to become eventually True. I don't know if there is already on this repo but we have a bunch of test utils that already do that.

Reply:
Honestly, I can`t think of one condition can be checked because it is a periodic cycle.

Reply:
You can replace a dependency you inject and attach a spy to it. I'll help with that

Reply:
What are we trying to wait for?

Why is this sleep here?

Reply:
The sleep is from the existing code, my understanding for it is to lazily wait for Kernel finish entire cycle then we can evaluate the result. This will be refactored, I will add a TODO.